apex/optimizers/fused_adam.py

```diff
@@ -144,15 +151,17 @@ def step(self, closure=None, grads=None, output_params=None, scale=None, grad_no
                 # State initialization
                 if len(state) == 0:
                     # Exponential moving average of gradient values
-                    state['exp_avg'] = torch.zeros_like(p.data)
+                    state['exp_avg'] = torch.zeros_like(p.data).float()
```

Comment:
I found that the m and v tensors also need to be in full precision for training to work properly, hence the changes here.